purchase_unreconciled/models/purchase_order.py

```diff
@@ -0,0 +1,278 @@
+            fields=["account_id", "product_id", "oca_purchase_line_id"],
+            groupby=["account_id", "product_id", "oca_purchase_line_id"],
+            lazy=False,
+        )
+        unreconciled_items = self.env["account.move.line"].search(unreconciled_domain)
+        for group in reconciling_groups:
+            account_id = group["account_id"][0]
+            product_id = group["product_id"][0] if group["product_id"] else False
+            purchase_line_id = (
+                group["oca_purchase_line_id"][0]
+                if group["oca_purchase_line_id"]
+                else False
+            )
+            unreconciled_items_group = unreconciled_items.filtered(
+                lambda line, account_id=account_id, product_id=product_id: (
+                    line.account_id.id == account_id
+                    and line.product_id.id == product_id
+                )
+            )
+            # Check which type of force reconciling we are doing:
+            # - Force reconciling amount_residual
+            # - Force reconciling amount_residual_currency
+            amount_residual_currency_reconcile = any(
+                unreconciled_items_group.filtered(
+                    lambda item_group,
+                    account_id=account_id: item_group.amount_residual_currency != 0.0
+                    and item_group.account_id.id == account_id
+                )
+            )
+            if amount_residual_currency_reconcile:
+                residual_field = "amount_residual_currency"
+            else:
+                residual_field = "amount_residual"
+            if float_is_zero(
+                sum(unreconciled_items_group.mapped(residual_field)),
+                precision_rounding=self.company_id.currency_id.rounding,
+            ):
+                moves_to_reconcile = unreconciled_items_group
+            else:
+                writeoff_vals = self._get_purchase_writeoff_vals(
+                    unreconciled_items_group, purchase_line_id, product_id
+                )
+                writeoff_to_reconcile = unreconciled_items_group._create_writeoff(
+                    writeoff_vals
+                )
+                all_writeoffs |= writeoff_to_reconcile
+                # add writeoff line to reconcile algorithm and finish the reconciliation
+                moves_to_reconcile = unreconciled_items_group | writeoff_to_reconcile
+            # Check if reconciliation is total or needs an exchange rate entry to be
+            # created
+            if moves_to_reconcile:
+                moves_to_reconcile.filtered(
+                    lambda move: not move.reconciled
+                ).reconcile()
+            reconciled_ids = unreconciled_items | all_writeoffs
+            res = {
+                "name": _("Reconciled journal items"),
+                "type": "ir.actions.act_window",
+                "view_type": "form",
+                "view_mode": "tree,form",
```

Comment:
```suggestion
                "view_mode": "list,form",
```

Getting this error, I think it is because of this line:

![image](https://github.com/user-attachments/assets/a8b46ebc-0334-4bd4-a7f4-13a9a4fafeb5)